metapool/prep.py

```diff
@@ -494,13 +440,36 @@ def preparations_for_run(run_path, sheet, generated_prep_columns,
             # this is the portion of the loop that creates the prep
             data = []
 
-            for well_id_col, sample in lane_sheet.iterrows():
-                if isinstance(sample, pd.core.series.Series):
-                    sample_id = well_id_col
-                else:
-                    sample_id = sample.sample_id
+            for sample_id, sample in lane_sheet.iterrows():
+                sample_name = sample.sample_name
+
+                if sample_id in unmapped:
+                    # sample_ids that couldn't be matched to a pair of fastq
+                    # files don't get entries in the prep file and therefore
+                    # further processing is skipped. These sample_ids are
+                    # recorded in the failed_samples_by_project dict, however.
+                    continue
+
+                associated_files = mapped[sample_id]
+
+                # run-prefix is a column value in the prep-info file and is
+                # programmatically defined as the part of the filename up to
+                # the orientation indicator (assuming it's present). e.g.:
+                # my_sample_S001_L001_R1_001.gz -> 'my_sample_S001_L001'.
 
-                run_prefix = get_run_prefix(run_path, project, sample_id, lane)
+                # we assume that the run-prefixes for all files
+                # associated with a sample are the same. Notify the user if
+                # this assertion isn't true as it's an indicator of a larger
+                # error.
+                run_prefixes = list(set([run_prefix_mapping[x] for x in
+                                         associated_files]))
+
+                if len(run_prefixes) != 1:
+                    raise ValueError(f'The files associated with {sample_name}'
+                                     f" ({', '.join(associated_files)}) do not"
+                                     " share the same run-prefix")
```

Comment:
Should we also test for len(run_prefixes) == len(lane_sheet.values())?

Reply:
not sure what you mean - a len_prefix in this case is just the first part of a sample-id/file-name up to the orientation notation ('R1', 'R2). What's important here is that the run_prefixes for the two files associated with the sample-id are identical. This is the code that generates prep-info files so there is no 'run_prefix' column in lane_sheet yet - this is the code that makes that column.

Reply:
I see, thank you for the explanation. Just for clarity, what I was asking was to check that the run_prefix for all the samples are unique; in other words that the len of the unique run_prefix and the unique samples are the same.